src/main/java/org/broadinstitute/hellbender/tools/walkers/conversion/GtfInfo.java

```diff
@@ -0,0 +1,55 @@
+package org.broadinstitute.hellbender.tools.walkers.conversion;
+
+import htsjdk.samtools.util.Interval;
+
+public class GtfInfo{
+
+    public enum Type{
+        GENE,
+        TRANSCRIPT
+    }
+
+    private Type type;
+    private final String geneName;
+    private Interval interval;
+
+    public GtfInfo(Interval interval, Type type, String geneName) {
+        this.interval = interval;
+        this.type = type;
+        this.geneName = geneName;
+    }
+
+    public Type getType() {
+        return type;
+    }
+
+    public void setType(Type type) {
+        this.type = type;
+    }
+
+    public String getGeneName() {
+        return geneName;
+    }
+
+    public Interval getInterval() {
+        return interval;
+    }
+
+    public Integer getStart() {
+        return interval.getStart();
+    }
+
+    public Integer getEnd() {
+        return interval.getEnd();
+    }
+
+    public void setInterval(Interval interval) {
```

Comment:
you are never using this to set the interval, instead you reconstruct this object everytime you need to change it. Both are valid apporaches but if the GTFInfo IS immutable you might consider using a java record class: https://docs.oracle.com/en/java/javase/17/language/records.html instead.

Reply:
Yes, final everywhere!